### PR TITLE
Recaptcha blog

### PR DESCRIPTION
**Pull request by tedk13**

### CODE REVIEW

File: src/content/posts/2024-08-27-adding-recaptcha-to-a-sign-up-form.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+image_credit: Glenn Carstens-Peters
+---
+
+Recently I was tasked with adding Google's reCAPTCHA to a couple of forms on one of our static sites. For what it's worth, our site is running on Astro. Just wanted to add that in. Anyway I have no experience with reCAPTCHA so I was thankful that a link to their docs was included in the issue. 
+
+You have some options when using reCAPTCHA. Versions 3 and 2. I opted for reCAPTCHA v2 Invisible. This prevents users from selecting fire hydrents or buses or whatever silly things one needs to click to prove they are a centient carbon based life form.
+
+## Adding Your Domain
+
+One of the first steps is [adding you domain](https://developers.google.com/recaptcha/docs/domain_validation) to your Admin Console in Google. In the console you will need to specify the version you want to use.
```
Comment (mrtrimble) [inline src/content/posts/2024-08-27-adding-recaptcha-to-a-sign-up-form.md:25]:
```suggestion
One of the first steps is [adding your domain](https://developers.google.com/recaptcha/docs/domain_validation) to your Admin Console in Google. In the console you will need to specify the version you want to use. 
```

**Review by mrtrimble — COMMENTED** [review]
very nice!!!

**Comment by tedk13** [comment]
> very nice!!!

thank you

**Review by mrtrimble — APPROVED** [review]

File: src/content/posts/2024-08-27-adding-recaptcha-to-a-sign-up-form.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+<script src="https://www.google.com/recaptcha/api.js?onload=onloadCallback&render=explicit"
+        async defer>
+</script>
+```
+
+You could simplify this even more. According to google, the simplest way is to [bind reCAPTCHA to a button](https://developers.google.com/recaptcha/docs/invisible#auto_render) on the form. For this approach you will need to add: 
+```html
+class="g-recaptcha" 
+data-sitekey="your_site_key"
```
Comment (eridolfi12) [inline src/content/posts/2024-08-27-adding-recaptcha-to-a-sign-up-form.md:68]: Is "your_site_key" supposed to be here?

Comment (tedk13) [reply src/content/posts/2024-08-27-adding-recaptcha-to-a-sign-up-form.md:68]: Yup. I didn't want to share the _actual_ key.

Comment (eridolfi12) [reply src/content/posts/2024-08-27-adding-recaptcha-to-a-sign-up-form.md:68]: @tedk13 I understand that but it looks like it was copied from the documentation and never updated.

Comment (tedk13) [reply src/content/posts/2024-08-27-adding-recaptcha-to-a-sign-up-form.md:68]: hahaha it was.

**Review by hougasian — APPROVED** [review]

**Pull request merged**